src/charm.py

```diff
@@ -69,59 +69,55 @@ def __init__(self, *args: Any) -> None:
     def exporters(self) -> List[BaseExporter]:
         """Return list of exporters based on detected hardware."""
         exporters: List[BaseExporter] = []
-        available_hw_tools = self.get_available_hw_tools_stored()
-        if available_hw_tools & HardwareExporter.hw_tools():
+        stored_tools = self.get_stored_tools()
+        if stored_tools & HardwareExporter.hw_tools():
             exporters.append(
                 HardwareExporter(
                     self.charm_dir,
                     self.model.config,
-                    available_hw_tools,
+                    stored_tools,
                 )
             )
 
-        if available_hw_tools & SmartCtlExporter.hw_tools():
+        if stored_tools & SmartCtlExporter.hw_tools():
             exporters.append(SmartCtlExporter(self.charm_dir, self.model.config))
 
         return exporters
 
-    def get_available_hw_tools_stored(self) -> Set[HWTool]:
-        """Get the available hardware tools from stored or from machine if not present.
+    def get_stored_tools(self) -> Set[HWTool]:
+        """Get the current hardware tools from stored or from machine if not present.
 
-        This function store the available hardware tools as string because HWTool object is not
-        accepted on Ops framework. However, to return the values it uses HWTool objects.
+        This function stores the current hardware tools as strings because StoredState cannot store
+        arbitrary objects. HWTool object can however be re-instantiated from tool names.
```

Comment:
Sorry my previous suggestion was not 100% correct
```suggestion
        arbitrary objects. HWTool objects can however be re-instantiated from tool names.
```